include/ozo/impl/read.h

```diff
@@ -24,6 +24,9 @@ inline Require<RawDataWritable<T>, istream&> read(istream& in, T& out) {
 template <typename T>
 inline Require<Integral<T> && sizeof(T) == 1, istream&> read(istream& in, T& out) {
     out = in.get();
+    if (!in) {
+        throw system_error(error::unexpected_eof);
```

Comment:
There is [exceptions](https://en.cppreference.com/w/cpp/io/basic_ios/exceptions) method for such stuff.

Reply:
Well, in case of using ```exception()``` method, ```std::ios_base::failure```  will be thrown which is not compatible with ```error_code``` and IMO completely meaningless - no useful info can be added at all. In case of throwing ```system_error``` (which is compatible with ```error_code``` by nature) additional string with information can be added in the future as soon as it needed.